### ISSUE TITLE
Longevity worker pods are evicted after 3 hours of longevity run

### ISSUE THREAD

**Issue by anishakj** [pravega/pravega#4366]
**Problem description**
While running longevity with `small-key` worker pods are getting evicted after 3hrs 50 mins.
```
root@anisha-VirtualBox:~# kubectl get po -n pravega-longevity
NAME                                READY   STATUS    RESTARTS   AGE
small-key-manager-554759b9b-9z9vv   1/1     Running   0          4h56m
small-key-worker-85d94bdc94-p5kzk   1/1     Running   0          33m
small-key-worker-85d94bdc94-ssclw   0/1     Evicted   0          4h56m
```
Looks like this issue is happening due to the regression of commit `Issue 4239: AppendProcessor pipelining (#4331)`: #4331 
In build previous to this commit `0.7.0-2411.c4e6dd8`, longevity is working fine with the same cluster.

**Problem location**
The regression has been detected after a change in `AppendProcessor`.

**Suggestions for an improvement**
Fix the problem.

**Comment by RaulGracia** [comment]
The cause for eviction seems to be memory usage, which is growing until the worker pod gets evicted:
![longevity-worker-pod-resources](https://user-images.githubusercontent.com/717112/68867503-36d6f500-06f6-11ea-8629-113707cf5c29.jpg)

**Comment by andreipaduroiu** [comment]
This is not a regression due to changes in `AppendProcessor`. The problem here is that your benchmark worker pod is evicted because it is running out of memory.

Why is this labeled as a `segmentstore` issue?

**Comment by RaulGracia** [comment]
Yes, I understand that @andreipaduroiu, but the point is that longevity workloads before #4331 do not experience this problem, and all the builds after that PR do experience this problem. While is just an hypothesis, we don't know if the interactions between the client and the `AppenProcessor` may have changed in a way that make the client to have a memory leak, so pods get evicted. Another option could be that now the client can write faster, so it consumes more memory because of the `AppendProcessor` improvement and then we need to limit the memory of the longevity pod (there is not `limit` defined for memory today). In any case, we wanted to keep track of this issued with the information we are gathering, and we will close the issue in the case we can fix this by setting proper resource limits to the pods.

**Comment by RaulGracia** [comment]
Thanks @andreipaduroiu, we are defining some reasonable resource `requests` and `limits` to pods and re-executing the experiment again. Will post here the result of that.

**Comment by anishakj** [comment]
I am still seeing the issue with the following resources set for worker pods
```
limits:
     cpu: 2
-    memory: 1024Mi
requests:
     cpu: 1
     memory:  500Mi
```

**Comment by RaulGracia** [comment]
@andreipaduroiu it looks like the longevity worker pod has been evicted due to lack of storage. That is, by default the log level is set to `DEBUG`, and after the change in `AppendProcessor` the worker pod generates around 10GB of logs per hours. As the internal storage for the pod is around 40/50GB, after 4 hours of execution the pod was being evicted consistently. @anishakj has changed the log level of the longevity application to `INFO`, and the current longevity has been working fine for +5 hours. According to @anishakj, most of the fast-growing logs at the client side were related to Netty. I don't know if it is required to revisit why after #4331 the client is producing so many logs, but at least we know the origin of the problem and how to mitigate the log overflow. Given that, I'm in favor of closing this issue and address the log level configuration of the longevity app in a separate issue.

**Comment by andreipaduroiu** [comment]
Why are we running longevity with DEBUG? I thought it should be INFO by default.

**Comment by RaulGracia** [comment]
@andreipaduroiu apparently the worker is set to `DEBUG` by default, which we may need to change, preferably making it parameterizable.
